File: 283-move-zeroes/move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        pos=0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[pos]=nums[i]
                pos=pos+1
        while(pos<len(nums)):
            nums[pos]=0
            pos=pos+1
        print(nums)
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int j=-1;
        int n=nums.length;
        for(int i=0;i<n;i++){
            if(nums[i]==0){
                j=i;
                break;
            }
        }
        if(j==-1) return;

        for(int k=j+1;k<n;k++){
            if(nums[k]!=0){
                int t=nums[k];
                nums[k]=nums[j];
                nums[j]=t;
                j++;
            }
        }
    }
}